### DISCUSSION TITLE
直接插入Bitmap数据

### DISCUSSION THREAD

**Discussion by doingf** [apache/doris#10459] [A - General / Q&A]
看文档描述BitMap类型的数据写入都需要通过to_bitmap()函数的定义转换，如果我想通过代码直接load进去bitmap类型的值, 不经过函数转换；使用Java代码请问应该怎么操作?测试通过Jdbc的setBytes()或setObject()都无法实现（借用了BitmapValue.java代码逻辑）

**Comment by FreeOnePlus** [comment]
当前 Doris 并没有提供关于 Bitmap 二进制类型的写入接口，有导出的接口，所以如果要插入 bitmap 类型的值，当前还是必须要走 to_bitmap() 函数进行转换

**Reply by FreeOnePlus** [reply]
还有一种方式，就是使用Spark Load的方式，详细可以看如下链接
http://doris.incubator.apache.org/zh-CN/docs/data-operate/import/import-way/spark-load-manual.html#hive-bitmap-udf

**Reply by doingf** [reply]
> 还有一种方式，就是使用Spark Load的方式，详细可以看如下链接 http://doris.incubator.apache.org/zh-CN/docs/data-operate/import/import-way/spark-load-manual.html#hive-bitmap-udf

我看Spark Load方式导入数据似乎也是要求Hive表，并配置字段映射为doris_column=bitmap_dict(hive_column)或者doris_column=binary_bitmap(hive_column)计算bitmap类型的值吧，所以这种方式也是bitmap实际是在doris端计算的？

**Reply by swordDL** [reply]
我也碰到这个问题了，我尝试jdbc连接，然后insert into xxx values (x,x,x,to_bitmap(x)）,这么做是可以的，但是官方因为性能问题，不建议这么写数据，如果你数据少可以这么来。

**Reply by doingf** [reply]
> 我也碰到这个问题了，我尝试jdbc连接，然后insert into xxx values (x,x,x,to_bitmap(x)）,这么做是可以的，但是官方因为性能问题，不建议这么写数据，如果你数据少可以这么来。

是想自己通过Java某些算法自己计算出bitmap值，然后插入进去，调用doris的to_bitmap()方法数据太多了感觉对服务压力大

**Reply by swordDL** [reply]
我的需求是一样的，但是目前看来没办法这么做，最终我还是通过任务调度工具，凌晨执行脚本，通过to_bitmap()写入数据，由doris自己计算。

**Reply by doingf** [reply]
> 我的需求是一样的，但是目前看来没办法这么做，最终我还是通过任务调度工具，凌晨执行脚本，通过to_bitmap()写入数据，由doris自己计算。

现有版本只能用服务提供的方法了，不知道后续能否自己计算插入

**Comment by niutaofan** [comment]
可以试试这种方式，我验证过了，没问题
https://github.com/apache/doris/pull/45231
